### ISSUE TITLE
Install instructions

### ISSUE THREAD

**Issue by MrMatch246** [jgoizueta/monologue#1]
Could you add some info on how to install? First time ruby user and it wants unimidi and liquid and the internet mentions gem files which i don't see any so i "guess" you might have had them already installed before like it happens in python too :D

**Comment by leshnik** [comment]
I also had problems running the program (`require': cannot load such file -- unimidi (LoadError) ). I'm thinking this might be because my Monologue is only connected via USB and not via MIDI. Solution was to edit _view_monologue.rb_ and comment out lines 2 & 3: 

`# require 'unimidi'`
`# require_relative 'midi_dump'`

Program then worked like a charm on individual program files saved from Korg's _monologue Sound Librairian_. Hope this helps. Thankyou @jgoizueta !

**Comment by diokomen** [comment]
Hi, I never used ruby and I don't really get how to get the scripts to work, could you please help me? Thanks

**Comment by leshnik** [comment]
Hi,
Do you have ruby installed? I'm using a macos system and installed it with homebrew. Once its installed, my instructions might help you. Excuse me if some of the steps are obvious to you.

I downloaded the zip file (the green Code button: Download ZIP (https://github.com/jgoizueta/monologue); decompress the file and in a terminal session cd into the directory "monologue_master". Open the file "view_monologue.rb" with a text editor and comment out lines 2 and 3 (# require 'unimidi')(# require_relative 'midi_dump'); then save the file.

The next step is done with the monologue librarian software and makes a dump of the contents. In my case that gave me a file with whatever I named it and the extension molglib, something like "2025-01-26.molglib".

Final step is to run the program on the command line (the terminal is still in the directory with the script): "ruby view_monologue.rb /Users/yourname/Documents/monologue\ library\ 2025-01-26.molglib". "ruby view_monologue.rb" is the command and it needs a .molglib file to operate on. In my case the file was in the Documents folder. You could put the output of the librarian software in the same directory as the script to make a first run easier.

When it runs, your browser opens with a nice graphic with the positions of all the knobs for each of the programs the librarian dumped.

I think that the monologue could communicate directly with the script without doing the librarian dump, but I didn't take the time to explore that. 

Hope this helps,
morton

**Comment by diokomen** [comment]
No, I got all the monologue-related stuff on windows, I'm not really into coding (as you can guess), was barely able to create monologue presets through reverse engineering factory ones and felt like a genius.

Thanks a lot for the exhaustive instructions, you're a legend!
